portal/views.py

```diff
@@ -0,0 +1,174 @@
+from django.core.exceptions import PermissionDenied
+from django.utils import timezone
+from rest_framework import generics, viewsets
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from portal.models import Poll, PollOption, PollVote
+from portal.serializers import (
+    AdminPollSerializer,
+    CreateUpdatePollVoteSerializer,
+    PollOptionSerializer,
+    RetrievePollSerializer,
+    RetrievePollVoteSerializer,
+    UserPollSerializer,
+)
+
+
+class RetrievePolls(viewsets.ReadOnlyModelViewSet):
+    """Viewset for browsing polls for user, and reviewing polls for admins"""
+
+    permission_classes = [IsAuthenticated]
+    queryset = Poll.objects.all()
+    serializer_class = RetrievePollSerializer
+
+    @action(detail=False, methods=["get"])
+    def browse(self, request):
+        # filters for all possible available polls, then filters for all polls answered
+        # by user, and finally returns the difference in poll sets
+        polls_available = Poll.objects.filter(expire_date__gte=timezone.localtime(), approved=True)
+        polls_answered = PollVote.objects.filter(
+            user=request.user, poll__in=polls_available
+        ).values_list("poll", flat=True)
+        return Response(
+            self.serializer_class(polls_available.exclude(id__in=polls_answered), many=True).data
+        )
+
+    @action(detail=False, methods=["get"])
+    def review(self, request):
+        # checks if user is admin
+        if not request.user.is_superuser:
+            raise PermissionDenied()
+        # returns list of polls that still need to be approved
+        return Response(self.serializer_class(Poll.objects.filter(approved=False), many=True).data)
```

Comment:
I think you can also do this by making a custom get_queryset function that returns different results depending on the view